abdiff/core/build_ab_images.py

```diff
@@ -1 +1,67 @@
-"""abdiff.core.build_ab_images"""
+import logging
+import os
+import shutil
+
+import docker
+from pygit2 import clone_repository
+from pygit2.enums import ResetMode
+
+from abdiff.core.utils import update_or_create_job_json
+
+logger = logging.getLogger(__name__)
+
+
+def build_ab_images(
+    job_directory: str,
+    docker_client: docker.client.DockerClient,
+    commit_sha_a: str,
+    commit_sha_b: str,
+) -> tuple[str, str]:
+    """Build Docker images based on 2 commit SHAs.
+
+    Args:
+        job_directory: The directory containing all files related to a job.
+        docker_client: A configured Docker client.
+        commit_sha_a: The SHA of the first commit for comparison.
+        commit_sha_b: The SHA of the second commit for comparison.
+    """
+    images = []
+    for commit_sha in [commit_sha_a, commit_sha_b]:
```

Comment:
What if we created a helper function to perform this work?  Then this parent/driver function is just responsible for organizing arguments and preparing things, but a little, terse function actually performs the clone + build?

Seems like it could be easier to test, but I haven't looked at the tests yet in detail.

Reply:
Depending on if, and how, this is refactored to a standlone function, I do think some logging could be helpful as well confirming that bits and pieces have been performed.  But highly dependent on if/how any changes are made during this PR.

Reply:
I don't have a strong preference myself, but I would like to keep the `images_update` and `update_or_create_job_json` _outside_ of the helper function such that -- the helper function is specific to _this command_ while the `update_or_create_job_json` is a universal step across all commands.

Reply:
If this helper function were created -- and big "if", unsure if needed -- I'd agree with @jonavellecuerdo that updating the job JSON would be outside its scope.  That feels like the work of the "core" function.

Reply:
Created a helper function, it does make the code more readable